src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java

```diff
@@ -654,13 +618,30 @@ public String getQuestionAdditionalInfoHtml() {
         return getQuestionDetails().getQuestionAdditionalInfoHtml(questionNumber, "");
     }
 
+    private static FeedbackQuestionDetails deserializeFeedbackQuestionDetails(String questionDetailsInJson,
+                                                                              FeedbackQuestionType questionType) {
+        if (questionType == FeedbackQuestionType.TEXT) {
+            return deserializeFeedbackTextQuestionDetails(questionDetailsInJson);
+        }
+        return JsonUtils.fromJson(questionDetailsInJson, questionType.getQuestionDetailsClass());
+    }
+
+    private static FeedbackQuestionDetails deserializeFeedbackTextQuestionDetails(String questionDetailsInJson) {
+        try {
+            // There are `FeedbackTextQuestion` with plain text, Json without `recommendedLength`, and complete Json
+            // in data store. Gson cannot parse the plain text case, so we need to handle it separately.
```

Comment:
@xpdavid this sounds like something that needs data migration.

Reply:
Yup. But it is not introduced in this PR. It a legacy pending data migration introduced long time ago :p